### PR TITLE
Additional user info prompts

### PR DESCRIPTION
**Pull request by allymreid**
- Added name prompt to profile creation page
- Added DOB to sign up page

- Added password/username criteria:
Username:
- Less than 24 char
- No whitespace
- No special characters except_

Password:
- Must contain number
- No whitespace
- 8-24 chars
- Must contain special character
- Must contain letter

### CODE REVIEW

**Review by AbdulTur — CHANGES_REQUESTED** [review]
Looks great. is it possible to add a change to the valid password function to also check if password contains atleast 1 letter and also contains a special character like, !@#*()_-+, otherwise looks good.

File: frontend/src/pages/UserSignup.js [hunk 1]
```diff
@@ -2,50 +2,78 @@ import React, { useState } from 'react';
...
-
-        navigate('/profile-creation');
-      } catch (error) {
-        if (error.response && error.response.data) {
-          setErrorMessage(error.response.data.error); 
-        } else {
-          setErrorMessage('Signup failed. Please try again.');
-        }
-      }
-
```
Comment (AbdulTur) [inline frontend/src/pages/UserSignup.js:44]: Also it looks like the api handling code was deleted. That needs to stay there

**Review by AbdulTur — CHANGES_REQUESTED** [review]

**Review by AbdulTur — APPROVED** [review]
LGTM Well done!!

**Review by WyrdWyn4 — APPROVED** [review]
LGTM, Great Job!

**Pull request merged**